Build failure on iOS

On iOS when I build this library, I get this error:

```
/Users/russellgreene/ars/ext/vcpkg_installed_ios/arm64-ios/include/vk_mem_alloc.h:6446:10: error: call to member function 'WriteNumber' is ambiguous
    json.WriteNumber(allocationCount);
    ~~~~~^~~~~~~~~~~
/Users/russellgreene/ars/ext/vcpkg_installed_ios/arm64-ios/include/vk_mem_alloc.h:5686:21: note: candidate function
void VmaJsonWriter::WriteNumber(uint32_t n)
                    ^
/Users/russellgreene/ars/ext/vcpkg_installed_ios/arm64-ios/include/vk_mem_alloc.h:5693:21: note: candidate function
void VmaJsonWriter::WriteNumber(uint64_t n)
                    ^
/Users/russellgreene/ars/ext/vcpkg_installed_ios/arm64-ios/include/vk_mem_alloc.h:6449:10: error: call to member function 'WriteNumber' is ambiguous
    json.WriteNumber(unusedRangeCount);
    ~~~~~^~~~~~~~~~~
/Users/russellgreene/ars/ext/vcpkg_installed_ios/arm64-ios/include/vk_mem_alloc.h:5686:21: note: candidate function
void VmaJsonWriter::WriteNumber(uint32_t n)
                    ^
/Users/russellgreene/ars/ext/vcpkg_installed_ios/arm64-ios/include/vk_mem_alloc.h:5693:21: note: candidate function
void VmaJsonWriter::WriteNumber(uint64_t n)
                    ^
/Users/russellgreene/ars/ext/vcpkg_installed_ios/arm64-ios/include/vk_mem_alloc.h:15967:34: error: call to member function 'ContinueString' is ambiguous
                            json.ContinueString(index++);
                            ~~~~~^~~~~~~~~~~~~~
/Users/russellgreene/ars/ext/vcpkg_installed_ios/arm64-ios/include/vk_mem_alloc.h:5657:21: note: candidate function
void VmaJsonWriter::ContinueString(uint32_t n)
                    ^
/Users/russellgreene/ars/ext/vcpkg_installed_ios/arm64-ios/include/vk_mem_alloc.h:5663:21: note: candidate function
void VmaJsonWriter::ContinueString(uint64_t n)
```

It seems like it's because size_t is not typedeff'd to either uint64_t or uint32_t. I'm not exactly sure what it is typdeff'd to, but I imagine an explicit cast would be a reasonable solution. 

If this seems fine, let me know and I'm happy to open a PR.

There is a workaround, which is defining
`#define VMA_STATS_STRING_ENABLED 0 `

--- a/include/vk_mem_alloc.h
+++ b/include/vk_mem_alloc.h
@@ -6468,10 +6468,10 @@ void VmaBlockMetadata::PrintDetailedMap_Begin(class VmaJsonWriter& json,
     json.WriteSize(unusedBytes);
 
     json.WriteString("Allocations");
-    json.WriteSize(allocationCount);
+    json.WriteSize((uint64_t)allocationCount);
 
     json.WriteString("UnusedRanges");
-    json.WriteSize(unusedRangeCount);
+    json.WriteSize((uint64_t)unusedRangeCount);
 
     json.WriteString("Suballocations");
     json.BeginArray();
@@ -16012,7 +16012,7 @@ void VmaAllocator_T::PrintDetailedMap(VmaJsonWriter& json)
                         {
                             json.WriteString("Name");
                             json.BeginString();
-                            json.ContinueString_Size(index++);
+                            json.ContinueString_Size((uint64_t)index++);
                             if (pool->GetName())
                             {
                                 json.ContinueString(" - ");

